frame/evm/src/lib.rs

```diff
@@ -520,6 +526,23 @@ pub type BalanceOf<T> =
 type NegativeImbalanceOf<C, T> =
 	<C as Currency<<T as frame_system::Config>::AccountId>>::NegativeImbalance;
 
+#[derive(Clone, Copy, Eq, PartialEq, Encode, Decode, TypeInfo)]
+pub struct CodeMetadata {
+	pub size: u64,
+	pub hash: H256,
+}
+
+impl CodeMetadata {
+	fn from_code(code: &[u8]) -> Self {
+		use sha3::Digest;
+
+		let size = code.len() as u64;
+		let hash = H256::from_slice(sha3::Keccak256::digest(&code).as_slice());
```

Comment:
If the account not have any code, we will compute a hash of an empty bytes vector at runtime, I wonder if it's the expected behavior.
If it's expected it's till unoptimized, it would be better to check if code.is_empty() and return an hardcoded constant in that case.